src/components/map/layers/ArrowDecorator.ts

```diff
@@ -0,0 +1,69 @@
+import { Path, PathProps, withLeaflet } from 'react-leaflet';
+import L, { PolylineDecorator, LatLng } from 'leaflet';
+import { LeafletContext } from '../Map';
+
+interface IArrowDecoratorProps extends PathProps {
+    leaflet: LeafletContext;
+    geometry: LatLng[];
+    onClick?: () => void;
+    color: string;
+    disableOnEventName?: string;
+    enableOnEventName?: string;
```

Comment:
Hmm, now when I think maybe "showOnEventName and hideOnEventName" would be better variable names